drivers: media: camera V2: Bringing back some of the mido camera specific changes to 4.9 from 3.18

offset1313 opened the pull request.

krasCGQ committed:
mido: ARM64: dts: Update qpnp-haptics configuration

* Basically applying 9e5ece273c44 to mido dtsi

Signed-off-by: Albert I <krascgq@outlook.co.id>

offset1313 committed:
Update msm_sensor_init.c

offset1313 committed:
drivers: media: camera v2; adapted for mido

offset1313 committed:
driver: media: camera v2: msm_isp40 adapted for mido

offset1313 committed:
drivers : media : camera v2: adapted for mido

offset1313 committed:
Merge pull request #1 from offset1313/patch-2

Patch 2

offset1313 committed:
Merge pull request #2 from offset1313/patch-3

drivers: media: camera v2; adapted for mido

offset1313 committed:
Merge pull request #3 from offset1313/patch-4

driver: media: camera v2: msm_isp40 adapted for mido

Kiciuk commented:
This commit:` drivers: media: camera v2; adapted for mido`
Brings some oem HW_INFO(Huaqin Sysfs Driver) driver hooks from downstream.
This thing is used by some OEM tools, we don't have such thing on this kernel and it will most likely gives an error when we try to compile kernel.
Imo it should be stashed,but im not dev here.

Nicklas373 committed:
drivers: misc: Import hqsysfs from Xiaomi

Signed-off-by: Dicky Herlambang <herlambangdicky5@gmail.com>

Nicklas373 commented:
> This commit:` drivers: media: camera v2; adapted for mido`
> Brings some oem HW_INFO(Huaqin Sysfs Driver) driver hooks from downstream.
> This thing is used by some OEM tools, we don't have such thing on this kernel and it will most likely gives an error when we try to compile kernel.
> Imo it should be stashed,but im not dev here.

Yes, but maybe we can try to import that hqsysfs from xiaomi merge for mido, i've found that xiaomi was merge and that code was available in that commit

offset1313 committed:
ASoC: fix snprintf size arguments

../techpack/audio/asoc/codecs/wsa881x.c:392:9: error: 'snprintf' size argument is too large; destination buffer has size 20, but size argument is 25 [-Werror,-Wfortify-source]
                len = snprintf(tmp_buf, 25, "0x%.3x: 0x%.2x\n", i,
                      ^
1 error generated.

The warnings are correct. Fix the size arguments by only allowing the
amount of bytes that the size of the char arrays are.

Signed-off-by: shagbag913 <sh4gbag913@gmail.com>
Signed-off-by: Khusika Dhamar Gusti <khusikadhamar@gmail.com>

offset1313 committed:
ASoC: fix sprintf warning in wcd_cpe_core

../techpack/audio/asoc/codecs/wcd_cpe_core.c:1981:2: error: 'snprintf' size argument is too large; destination buffer has size 11, but size argument is 13 [-Werror,-Wfortify-source]
        snprintf(proc_name, (sizeof("cpe") + sizeof("_state") +
        ^
1 error generated.

nE0sIghT committed:
mido: audio: fix audio cracks

This nice finding was done by @lostark13 and published at [1]. Let's reuse
this change but reduce sleep range and guard it by CONFIG_MACH_XIAOMI_MIDO

[1] https://github.com/lostark13/android_kernel_xiaomi_msm8953/commit/91cf3ba1b11b5f5d87d70b55e48a1f37ee180751

See-Also: https://github.com/nE0sIghT/android_kernel_xiaomi_msm8953/issues/11

offset1313 committed:
Merge pull request #4 from nE0sIghT/lineage-16.0-mido-4.9

sync

offset1313 committed:
audio : asoc :  Don't pass endianess and mode as argument to pcm format v3 function

Endianess and mode should not be passed as arguments to q6asm_media_format_block_pcm_format_support_v3  function
Hence following error has been fixed:
../techpack/audio/asoc/msm-compress-q6-v2.c:1046:6: error: too many arguments to function call, expected 8, have 10
                                        ASM_LITTLE_ENDIAN,
                                        ^~~~~~~~~~~~~~~~~~
../techpack/audio/include/dsp/q6asm-v2.h:111:27: note: expanded from macro 'ASM_LITTLE_ENDIAN'
#define ASM_LITTLE_ENDIAN 0
                          ^
../techpack/audio/include/dsp/q6asm-v2.h:510:5: note: 'q6asm_media_format_block_pcm_format_support_v3' declared here
int q6asm_media_format_block_pcm_format_support_v3(struct audio_client *ac,
    ^
1 error generated.
